bridge/device/cloud/manager.go

```diff
@@ -223,36 +220,29 @@ func (c *Manager) getCloudConfiguration() Configuration {
 	return c.private.cfg
 }
 
-func (c *Manager) setCreds(creds CoapSignUpResponse) {
-	c.creds = creds
-	if creds.ExpiresIn != 0 {
-		c.creds.ValidUntil = time.Now().Add(time.Duration(creds.ExpiresIn) * time.Second)
-	}
-	c.signedIn = false
-}
-
-func (c *Manager) updateCredsBySignInResponse(resp CoapSignInResponse) {
-	c.creds.ExpiresIn = resp.ExpiresIn
-	if resp.ExpiresIn != 0 {
-		c.creds.ValidUntil = time.Now().Add(time.Duration(resp.ExpiresIn) * time.Second)
+func validUntil(expiresIn int64) time.Time {
+	if expiresIn == -1 {
+		return time.Time{}
 	}
-	c.signedIn = true
+	return time.Now().Add(time.Duration(expiresIn) * time.Second)
 }
 
-func (c *Manager) updateCredsByRefreshTokenResponse(resp CoapRefreshTokenResponse) {
-	c.creds.AccessToken = resp.AccessToken
-	c.creds.RefreshToken = resp.RefreshToken
-	c.creds.ExpiresIn = resp.ExpiresIn
-	if resp.ExpiresIn != 0 {
-		c.creds.ValidUntil = time.Now().Add(time.Duration(resp.ExpiresIn) * time.Second)
-	}
+func (c *Manager) setCreds(creds ocfCloud.CoapSignUpResponse) {
+	c.creds = creds
 	c.signedIn = false
 }
 
-func (c *Manager) getCreds() CoapSignUpResponse {
+func (c *Manager) getCreds() ocfCloud.CoapSignUpResponse {
 	return c.creds
 }
 
+func (c *Manager) isCredsExpiring() bool {
+	if !c.signedIn || c.creds.ValidUntil.IsZero() {
+		return false
+	}
+	return !time.Now().Before(c.creds.ValidUntil.Add(-time.Second * 10))
```

Comment:
need to be calculated:
 - minimum value 10sec before expires
 - the first refresh occurs in 2/3 valid until